spec/unit/provider/osx_profile_spec.rb

```diff
@@ -0,0 +1,259 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Chef::Provider::OsxProfile do
+  let(:shell_out_success) do
+    double('shell_out', :exitstatus => 0, :error? => false)
+  end
+  describe 'action_create' do
+    before(:each) do
+      @node = Chef::Node.new
```

Comment:
D'oh.  I should have posted this earlier but just realized.

The desired way of writing RSpec tests is to use `let` instead of making instance variables.  This `before(:each)` block should be rewritten as:

```ruby
let(:node) { Chef::Node.new }
let(:events) { Chef::EventDispatch::Dispatcher.new }
let(:run_context) { Chef::RunContext.new(node, {}, events) }
let(:new_resource) { Chef::Resource::OsxProfile.new("Profile Test", run_context) }
let(:provider) { Chef::Provider::OsxProfile.new(new_resource, run_context) }
let(:all_profiles) do
  {"_computerlevel"=>
      [{"ProfileDisplayName"=>"Finder Settings",
        "ProfileIdentifier"=>"com.apple.finder",
        "ProfileInstallDate"=>"2015-11-08 23:15:21 +0000",
        "ProfileItems"=>
         [{"PayloadContent"=>
            {"PayloadContentManagedPreferences"=>
              {"com.apple.finder"=>
                {"Forced"=>[{"mcx_preference_settings"=>{"ShowExternalHardDrivesOnDesktop"=>false}}]}}},
           "PayloadDisplayName"=>"Custom: (com.apple.finder)",
           "PayloadIdentifier"=>"com.apple.finder",
           "PayloadType"=>"com.apple.ManagedClient.preferences",
           "PayloadUUID"=>"a017048f-684b-4e81-baa3-43afe316d739",
           "PayloadVersion"=>1}],
        "ProfileOrganization"=>"Facebook",
        "ProfileRemovalDisallowed"=>"false",
        "ProfileType"=>"Configuration",
        "ProfileUUID"=>"e2e09bef-e673-44a6-bcbe-ecb5f1c1b740",
        "ProfileVerificationState"=>"unsigned",
        "ProfileVersion"=>1},
      {"ProfileDisplayName"=>"ScreenSaver Settings",
        "ProfileIdentifier"=>"com.testprofile.screensaver",
        "ProfileInstallDate"=>"2015-10-05 23:15:21 +0000",
        "ProfileItems"=>
         [{"PayloadContent"=>
            {"PayloadContentManagedPreferences"=>
              {"com.apple.screensaver"=>
                {"Forced"=>[{"mcx_preference_settings"=>{"idleTime"=>0}}]}}},
           "PayloadDisplayName"=>"Custom: (com.apple.screensaver)",
           "PayloadIdentifier"=>"com.apple.screensaver",
           "PayloadType"=>"com.apple.ManagedClient.preferences",
           "PayloadUUID"=>"73fc30e0-1e57-0131-c32d-000c2944c110",
           "PayloadVersion"=>1}],
        "ProfileOrganization"=>"Facebook",
        "ProfileRemovalDisallowed"=>"false",
        "ProfileType"=>"Configuration",
        "ProfileUUID"=>"1781fbec-3325-565f-9022-8aa28135c3cc",
        "ProfileVerificationState"=>"unsigned",
        "ProfileVersion"=>1}],
  }
end

let(:test_profile) do
  {
  'PayloadIdentifier' => 'com.testprofile.screensaver',
  'PayloadRemovalDisallowed' => false,
  'PayloadScope' => 'System',
  'PayloadType' => 'Configuration',
  'PayloadUUID' => '1781fbec-3325-565f-9022-8aa28135c3cc',
  'PayloadOrganization' => 'Chef',
  'PayloadVersion' => 1,
  'PayloadDisplayName' => 'Screensaver Settings',
  'PayloadContent'=> [
    {
      'PayloadType' => 'com.apple.ManagedClient.preferences',
      'PayloadVersion' => 1,
      'PayloadIdentifier' => 'com.testprofile.screensaver',
      'PayloadUUID' => '73fc30e0-1e57-0131-c32d-000c2944c108',
      'PayloadEnabled' => true,
      'PayloadDisplayName' => 'com.apple.screensaver',
      'PayloadContent' => {
        'com.apple.screensaver' => {
          'Forced' => [
            {
              'mcx_preference_settings' => {
                'idleTime' => 0,
              }
            }
          ]
        }
      }
    }
  ]
  }
end
let(:no_profiles) do
  { "_computerlevel"=> [] }
end

before(:each) do
  allow(provider).to receive(:cookbook_file_available?).and_return(true)
  allow(provider).to receive(:cache_cookbook_profile).and_return('/tmp/test.mobileconfig.remote')
  allow(provider).to receive(:get_new_profile_hash).and_return(test_profile)
  allow(provider).to receive(:get_installed_profiles).and_return(all_profiles)
  allow(provider).to receive(:read_plist).and_return(all_profiles)
  allow(::File).to receive(:unlink).and_return(true)
end
```

Then you can replace all references to `@new_resource` with `new_resource`, `@provider` with `provider`, etc. in the examples and context blocks below.